Adding new Scala and Java benchmarks

marcospereira opened the pull request:
# Dependencies

This depends on #5 and #6.

## Purpose

This will activate new Java and Scala benchmarks. Two new Lua scripts were added to support the new benchmarks. Apparently, wrk is capable of handling more than one script, so we can have something like `wrk ... -s wrk_report.lua -s wrk_post.lua ...`.

marcospereira committed:
Adding new Scala and Java benchmarks

gmethvin commented on README.md:
```diff
@@ -36,6 +44,14 @@ Test an action that returns a short HTML page generated from a template that tak
+
+1. Using Netty server
+2. Using [Akka HTTP](https://github.com/akka/akka-http) Server with default filters
+3. Using [Akka HTTP](https://github.com/akka/akka-http) Server **without** default filtersca
```
typo at the end

marcospereira replied:
Fixed. :-)

marcospereira committed:
Register new benchmarks

marcospereira commented:
@richdougherty this is ready to be reviewed.

Keep in mind that we first need to nail down #5 and #6.

Also, I have noticed that we don't have all possible benchmarks turned on. For example, we don't have `scala-minimal-template-simple`, but only `scala-di-template-simple`. That would explain why some of the graphs are empty.

Finally, I've made the changes only to `apps-master-10`. Please let me know if we need to also port these to `apps-master-8`.

richdougherty commented:
The `minimal` apps have the default filters disabled, including the CSRF filter. The form upload tests shouldn't check for a CSRF token.

For the other apps where the CSRF filter is still enabled (`di`, `netty`, etc) should we set up the form tests so we supply a real CSRF token?, i.e. have wrk supply a cookie value.

marcospereira commented:
> For the other apps where the CSRF filter is still enabled (di, netty, etc) should we set up the form tests so we supply a real CSRF token?, i.e. have wrk supply a cookie value.

It is possible, I think. But we should maybe do that in another PR. :-)

richdougherty commented:
Another PR sounds fine to me. :)

marcospereira committed:
Fix typo

marcospereira committed:
Activate all available tests for Scala and Java

For di, netty and minimal. Some of the tests were listed only for di applications.
